File: dagger2-scopes-and-subcomponents-part3/app/src/main/java/com/example/dagger2scopesandsubcomponentspart3/WarriorScreenModule.java
package com.example.dagger2scopesandsubcomponentspart3;


import dagger.Module;
import dagger.Provides;

@Module
public class WarriorScreenModule {

    @Provides
    @WarriorScreenScope
    WarriorPresenter warriorPresenter(Warrior warrior) {
        return new WarriorPresenter(warrior);
    }
}

File: dagger2-scopes-and-subcomponents2/app/src/main/java/com/example/dagger2scopesandsubcomponentspart2/WarriorScreenModule.kt
package com.example.dagger2scopesandsubcomponentspart2

import dagger.Module
import dagger.Provides

@Module
class WarriorScreenModule {

    @Provides
    @WarriorScreenScope
    fun provideWarriorPresenter(warrior : Warrior): WarriorPresenter {
        return WarriorPresenter(warrior)
    }
}